embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/aei/AeiDataSourceImpl.kt

```diff
@@ -0,0 +1,316 @@
+package io.embrace.android.embracesdk.capture.aei
+
+import android.app.ActivityManager
+import android.app.ApplicationExitInfo
+import android.os.Build.VERSION_CODES
+import androidx.annotation.RequiresApi
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.arch.datasource.LogDataSourceImpl
+import io.embrace.android.embracesdk.arch.destination.LogEventData
+import io.embrace.android.embracesdk.arch.destination.LogEventMapper
+import io.embrace.android.embracesdk.arch.destination.LogWriter
+import io.embrace.android.embracesdk.arch.limits.UpToLimitStrategy
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
+import io.embrace.android.embracesdk.capture.user.UserService
+import io.embrace.android.embracesdk.config.ConfigService
+import io.embrace.android.embracesdk.config.behavior.AppExitInfoBehavior
+import io.embrace.android.embracesdk.internal.utils.BuildVersionChecker
+import io.embrace.android.embracesdk.internal.utils.VersionChecker
+import io.embrace.android.embracesdk.internal.utils.toNonNullMap
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logDebug
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logInfoWithException
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logWarningWithException
+import io.embrace.android.embracesdk.payload.AppExitInfoData
+import io.embrace.android.embracesdk.payload.BlobMessage
+import io.embrace.android.embracesdk.payload.BlobSession
+import io.embrace.android.embracesdk.prefs.PreferencesService
+import io.embrace.android.embracesdk.session.id.SessionIdTracker
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import java.io.IOException
+import java.util.concurrent.Future
+import java.util.concurrent.atomic.AtomicBoolean
+
+@RequiresApi(VERSION_CODES.R)
+internal class AeiDataSourceImpl(
+    private val backgroundWorker: BackgroundWorker,
+    private val configService: ConfigService,
```

Comment:
I thought we had wanted to make the orchestrator push config info to the data sources rather than have them directly depend on them. Is the current state due to how everything is hooked up to the legacy stuff right now, or will this be necessary going forward?

Reply:
The orchestrator does control whether a feature is enabled/disabled but that's it right now.

There isn't any such approach for config that alters limits/how data is captured. IMO just passing a `ConfigService` or even a `*Behavior` class is probably the simplest option